Makefile

```
@@ -0,0 +1,71 @@
+# Go parameters
+GOCMD=go
+GOBUILD=$(GOCMD) build
+GORUN=$(GOCMD) run
+GOCLEAN=$(GOCMD) clean
+GOTEST=$(GOCMD) test
+GOMOD=$(GOCMD) mod
+
+# Source parameters
+SOURCE_ENTRYPOINT=./cmd/main.go
+
+# Binary parameters
+BINARY_NAME=tinyhen-server
+BINARY_DESTINATION=./bin
+BINARY_PATH=$(BINARY_DESTINATION)/$(BINARY_NAME)
+
+# Docker parameters
+DOCKERCMD=docker
+DOCKERBBUILD=$(DOCKERCMD) build
+DOCKERRUN=$(DOCKERCMD) run
+DOCKERSTOP=$(DOCKERCMD) stop
+DOCKERRM=$(DOCKERCMD) rm
+
+# Development
+tidy:
+		$(GOMOD) tidy
+build:
+		$(GOBUILD) -o $(BINARY_PATH) -v $(SOURCE_ENTRYPOINT)
+run:
+		$(GORUN) $(SOURCE_ENTRYPOINT)
+unit_test:
+		$(GOTEST) -v ./... -coverprofile=coverage.txt -covermode=atomic
+clean:
+		$(GOCLEAN) $(SOURCE_ENTRYPOINT)
+		rm -f $(BINARY_PATH)
+
+# Docker for local operations
+build-linux:
+		CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GOBUILD) -o $(BINARY_PATH)-amd64 -v $(SOURCE_ENTRYPOINT)
+docker_build: build-linux
+		$(DOCKERBBUILD) -t tinyhen-server .
+docker_run: build-linux docker_build
+		$(DOCKERRUN) -d -p 10010:10010 tinyhen-server
+docker_stop: docker_run
+		$(DOCKERSTOP) tinyhen-server
+docker_rm: docker_run
```

I find these prerequisites harsh. If someone types, by mistake `make docker_rm`, this will build_linux + docker_build + docker_run, when, maybe, we didn't need all this. But that's just me - shutting down, in my opinion, should not switch on all the lights. If the image doesn't exist, so be it, ignore the error DOCKERRM returns.

---

The previous comment was valid for stop and rm, and I think the rm simply cannot work, as it it : we try to docker rm a running container - docker will require -f and not remove it. (I suppose this is a typo, `docker_rm` should require `docker_stop`, not `run`)

---

I think that I wrote it but did not test it from scratch anyway. Not sure I will keep it.